### ISSUE TITLE
[Feature Request] - Support for "Other versions", Run on schedule, etc

### ISSUE THREAD

**Issue by Investigamer** [TheWicklowWolf/LidaTube#6]
## Prologue
Howdy! First of all, I'm extremely happy to stumble onto the project, and wanted to thank the owner and any other contributors for setting this up. Had this idea recently to use YouTube music as a fallback for missing stuff on lidarr and was super relieved to find that it already existed in a working state (previously was doing this manually via MeTube). Also happy to see the project is primarily Python, so I can actually help contribute since that's my bread and butter.

## Feature Request
So as we all know, music data across platforms so rarely agrees with one another, in this case if you look at an artist like **P.O.S.** the album "Chill, Dummy" is separate from the album "Chill, Dummy (Instrumental)" where lidarr is concerned (such is the same for many artists I'm sure). Ironically on Lidarr a different album for the **same artist** represents the instrumental version as a different release within the same album, rather than a totally separate album, but I digress.

In the case of the normal and instrumental versions of "Chill, Dummy", I was unable to download the instrumental through LidaTube since it presents on YouTube Music as a "Other version" and despite the name searched via LidaTube being the instrumental version it just corrects to the non-instrumental. So I had to manually link that instrumental album in MeTube to grab it. You can see the original album [here](https://music.youtube.com/playlist?list=OLAK5uy_lK5DQar5098wTv0WYKn2raVtL1jXcCSY0) and the instrumental "other version" [here](https://music.youtube.com/playlist?list=OLAK5uy_kzujpYZCOjyCQlZ8bQ160xaLzz_PUK4B8).

I think we should add the capability to pull in data for "Other versions" and look for closer matches in "Other versions" when searching for wholesale albums. I would absolutely be willing to commit some time to a PR for such a feature and will reference this issue in the PR if I get around to it soon! 😄 

## Additional Features
I see that some features previously requested have been marked "out of scope" for the project, which is understandable and I don't wish to step on anyone's toes here, but one feature in particular caught my eye, the ability to search individual songs if someone has part of an album but missing a song or two. I definitely understand not letting the project scope expand out of control and keeping things grounded and straightforward without too much code debt, but as I said I'm very experienced with Python (and JS) and would be willing to commit additional time to getting this feature implemented as well. I think it could be an excellent addition, but I just wanted to get clarification on that, if I were to work on such a feature and submit a competent PR to implement it, would the PR be reviewed and accepted if adequate, or would it still be out of scope? No pressure at all if the issue just isn't desired, just wanted to ask first before I commit the time to implementing that 😁

Also, I'm not sure if this has already been implemented, but I think adding some kind of daemon mode or lidarr hook that automatically grabs from YouTube when attempts from lidarr have been exhausted would be fantastic, or at least the ability to toggle the behavior of running a LidaTube search on a routine schedule. Personally I plan to use LidaTube to grab anything missing in my collection, then enable upgrades in Lidarr that way the YouTube version is always the fallback solution for music but better versions i.e. FLAC can be ingested if and when they become available.

## Unraid Template
Lastly, I think we should try to get an Unraid template listed in Unraid community apps. Would make a wonderful addition, since as of now there are no community apps that support this level of automation for grabbing music from YouTube as a fallback solution for lidarr that implement music metadata as competently as LidaTube does (well done on that!). As far as I can tell MeTube is the closest, but it has many drawbacks, mainly that you have to manually link every album you want, and the metadata is hit or miss.

**Comment by TheWicklowWolf** [comment]
It was out of scope for me when I set this up because I intended it to be a blunt instrument. 

Given the inconsistencies between music databases, I accepted that it wouldn't be able to retrieve everything, and some content would need to be pulled in manually. Therefore, I focused on getting as much content as possible without delving too deeply into it. 

This approach might not make the code ideal for expansion, but I'm definitely open to pull requests. Here's the way I see the priority list:

1. Improve matching
2. Search individual songs
3. Schedule functionality
4. Support for "Other versions" instead of entire albums

Regarding Unraid, I believe it's already available:
https://unraid.net/community/apps?q=lidatube#r

Ideally, this tool would be a plugin for Lidarr. However, that would require implementing it with APIs similar to those of a torrent client and an indexer for searching. I'm hopeful that someone will take on that challenge!

In essence, I aim to keep this tool as simple as possible, with a basic feature set, to minimize maintenance efforts. 
So, in summary, yes, I'm open to pull requests, but I aim to keep it fairly basic. 

Thanks for your interest
